app/web/src/components/EncodedUrl/helpers.ts

```diff
@@ -0,0 +1,75 @@
+import { useEffect } from 'react'
+import { flow } from 'lodash/fp'
+
+import { useIdeContext } from 'src/helpers/hooks/useIdeContext'
+import { useRender } from 'src/components/IdeWrapper/useRender'
+import { encode, decode } from 'src/helpers/compress'
+import { isBrowser } from '@redwoodjs/prerender/browserUtils'
+
+const scriptKey = 'encoded_script'
+const scriptKeyV2 = 'encoded_script_v2'
+const fetchText = 'fetch_text_v1'
+
+export const githubSafe = (url: string): string =>
+  url.includes('github.com')
+    ? url
+        .replace('github.com', 'raw.githubusercontent.com')
+        .replace('/blob/', '/')
+    : url
+
+export const prepareEncodedUrl = flow(decodeURIComponent, githubSafe)
+
+const prepareDecodedUrl = flow(githubSafe, encodeURIComponent)
+
+export function makeEncodedLink(code: string): string {
+  const encodedScript = encode(code)
+  return `${location.origin}${location.pathname}#${scriptKeyV2}=${encodedScript}`
+}
+
+export function makeExternalUrl(resourceUrl: string): string {
+  return `${location.origin}${
+    location.pathname
+  }#${fetchText}=${prepareDecodedUrl(resourceUrl)}`
+}
+
+export function useIdeInit(cadPackage: string) {
```

Comment:
All of the logic in this hook I had in the `IdeToolbarNew` because that was the main wrapping component that had the context provider, so it made sense to have init useEffects there. However since most of the logic relates to checking if there is something encoded in the URL, I thought it made sense to bundle the logic here with the rest of the encoded url code, and bring it into `IdeToolbarNew` as a custom hook.

I've also now rename `IdeToolbarNew` to `IdeWrapper`